src/services/refactors/extractMethod.ts

```diff
@@ -0,0 +1,885 @@
+/// <reference path="../refactorProvider.ts" />
+/// <reference path="../../compiler/checker.ts" />
+
+/* @internal */
+namespace ts.refactor.extractMethod {
+    export const name = "extract_method";
+    const extractMethod: Refactor = {
+        name: "Extract Method",
+        description: Diagnostics.Extract_function.message,
+        getAvailableActions,
+        getEditsForAction,
+    };
+
+    registerRefactor(extractMethod);
+
+    /** Compute the associated code actions */
+    function getAvailableActions(context: RefactorContext): ApplicableRefactorInfo[] | undefined {

```

Comment:
@amcasey this is an implementation of [`Refactor#getEditsForAction`](https://github.com/Microsoft/TypeScript/pull/16960/files#diff-bdb460eb990c9d2f8a0f5caf3b2f4d54R11)